### PR TITLE
Upload main images to Arcus S3 and sync clouds

### PR DESCRIPTION
**Pull request by bertiethorpe**
1st job cleans all images in Arcus S3 openhpc-images-prerelease bucket

2nd job uploads main branch fatimage to Arcus S3 openhpc-images-prerelease bucket when bumped.

3rd job retrieves image from S3 and saves to remianing ci cloud openstacks.

### CODE REVIEW

**Comment by bertiethorpe** [comment]
Bucket upload and image sync test: 
https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11275944018/job/31359713181

**Comment by bertiethorpe** [comment]
Bucket upload and image sync test:
https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11292572641/job/31410255776

File: .github/workflows/s3-image-sync.yml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+          TARGET_IMAGE=$(jq --arg version "${{ matrix.build }}" -r '.cluster_image[$version]' "${{ env.IMAGE_PATH }}")
+          echo "TARGET_IMAGE=${TARGET_IMAGE}" >> "$GITHUB_ENV"
+
+      - name: Download latest image if missing
+        run: |
+          . venv/bin/activate
+          bash .github/bin/get-s3-image.sh ${{ env.TARGET_IMAGE }} ${{ env.S3_BUCKET }}
+
+      - name: Cleanup OpenStack Image (on error or cancellation)
+        if: cancelled()
```
Comment (sd109) [inline .github/workflows/s3-image-sync.yml:152]: If this is supposed to run on errors too then I think we need this:
```suggestion
        if: cancelled() || failure()
```

File: .github/workflows/s3-image-sync.yml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+      OS_CLOUD: openstack
+      CI_CLOUD: ${{ matrix.cloud }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Record which cloud CI is running on
+        run: |
+          echo CI_CLOUD: ${{ env.CI_CLOUD }}
+
+      - name: setup environment
```
Comment (sd109) [inline .github/workflows/s3-image-sync.yml:127]: Nit:
```suggestion
      - name: Setup environment
```

File: .github/workflows/s3-image-sync.yml [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        cloud:
+          - LEAFCLOUD
+          - SMS
+          - ARCUS
+        build:
+          - RL8
+          - RL9
+          - RL9-cuda
+        exclude: 
+          - cloud: LEAFCLOUD
```
Comment (sd109) [inline .github/workflows/s3-image-sync.yml:114]: Why do we hard code leafcloud here? Is that assuming the original build will always run on leafcloud?

File: .github/workflows/s3-image-sync.yml [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+      OS_CLOUD: openstack
+      CI_CLOUD: ${{ vars.CI_CLOUD }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Record which cloud CI is running on
+        run: |
+          echo CI_CLOUD: ${{ env.CI_CLOUD }}
+
+      - name: setup environment
```
Comment (sd109) [inline .github/workflows/s3-image-sync.yml:57]: Nit:
```suggestion
      - name: Setup environment
```

File: .github/workflows/s3-image-sync.yml [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+          echo "${{ secrets['ARCUS_S3_CFG'] }}" > ~/.s3cfg
+        shell: bash
+
+      - name: Install s3cmd
+        run: |
+          sudo apt-get --yes install s3cmd
+      
+      - name: Cleanup S3 bucket
+        run: |
+          s3cmd rm s3://${{ env.S3_BUCKET }} --recursive --force
```
Comment (sd109) [inline .github/workflows/s3-image-sync.yml:33]: Any particular reason we need `--force` here?

File: .github/workflows/s3-image-sync.yml (hunk 3 above)
Comment (bertiethorpe) [reply .github/workflows/s3-image-sync.yml:114]: Actions don't allow dynamic setting of matrix values with vars I think, they only allow use of previous job outputs. We could go about it that way?

File: .github/workflows/s3-image-sync.yml (hunk 5 above)
Comment (bertiethorpe) [reply .github/workflows/s3-image-sync.yml:33]: Suppresses confirmation prompts

**Comment by bertiethorpe** [comment]
Bucket upload and image sync test:
https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11326425545/job/31502460548

**Review by sd109 — APPROVED** [review]
LGTM, nice work!

**Pull request merged**